src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -1408,6 +1399,22 @@ bool InterpreterCreateQuery::doCreateTable(ASTCreateQuery & create,
         addColumnsDescriptionToCreateQueryIfNecessary(query_ptr->as<ASTCreateQuery &>(), res);
     }
 
+    std::unordered_set<std::string_view> all_columns;
+    all_columns.reserve(properties.columns.size());
+
+    for (const auto & column : properties.columns)
+        all_columns.emplace(column.name);
+
+    auto reserved_columns = res->getVirtuals().getNames();
+    for (const auto & column : reserved_columns)
+    {
+        if (all_columns.contains(column))
+            throw Exception(ErrorCodes::ILLEGAL_COLUMN,
```

Comment:
What if it's attach? This will cause problems on startup if users already have tables created with virtual columns no?